platform-api/src/main/java/io/datacater/core/deployment/K8Deployment.java

```diff
@@ -255,6 +255,15 @@ public Map<String, Object> getDeployment(UUID deploymentId) {
     }
   }
 
+  public Deployment getDeploymentObject(UUID deploymentId) {
+    return client
+        .apps()
+        .deployments()
+        .inNamespace(StaticConfig.EnvironmentVariables.NAMESPACE)
+        .withName(getDeploymentName(deploymentId))
+        .get();
+  }
+
```

Comment:
by adding this method, we have redundant code in the `getDeployment()` method. Could you edit the getDeployment() to use this method ([K8Deployment.java Lines 244-249](https://github.com/DataCater/datacater/blob/main/platform-api/src/main/java/io/datacater/core/deployment/K8Deployment.java#L244-L249))?